x-pack/qa/sql/src/main/java/org/elasticsearch/xpack/qa/sql/jdbc/ResultSetBaseTestCase.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.qa.sql.jdbc;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.common.CheckedConsumer;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.xpack.sql.jdbc.jdbc.JdbcConfiguration;
+import org.elasticsearch.xpack.sql.jdbc.jdbcx.JdbcDataSource;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+abstract class ResultSetBaseTestCase extends JdbcIntegrationTestCase {
+    
+    static final Set<String> fieldsNames = Stream.of("test_byte", "test_integer", "test_long", "test_short", "test_double",
+            "test_float", "test_keyword")
+            .collect(Collectors.toCollection(HashSet::new));
+    static final List<String> dateTimeTestingFields = Stream.of("test_boolean", "test_byte", "test_integer", "test_long",
+            "test_short", "test_double", "test_float", "test_keyword").collect(Collectors.toCollection(ArrayList::new));
+    static final String SELECT_ALL_FIELDS = "SELECT test_boolean, test_byte, test_integer,"
+            + "test_long, test_short, test_double, test_float, test_keyword, test_date FROM test";
+    static final String SELECT_WILDCARD = "SELECT * FROM test";
+
+    protected void createIndex(String index) throws Exception {
+        Request request = new Request("PUT", "/" + index);
+        XContentBuilder createIndex = JsonXContent.contentBuilder().startObject();
+        createIndex.startObject("settings");
+        {
+            createIndex.field("number_of_shards", 1);
+            createIndex.field("number_of_replicas", 1);
+        }
+        createIndex.endObject();
+        createIndex.startObject("mappings");
+        {
+            createIndex.startObject("doc");
+            {
+                createIndex.startObject("properties");
+                {}
+                createIndex.endObject();
+            }
+            createIndex.endObject();
+        }
+        createIndex.endObject().endObject();
+        request.setJsonEntity(Strings.toString(createIndex));
+        client().performRequest(request);
+    }
+
+    protected void updateMapping(String index, CheckedConsumer<XContentBuilder, IOException> body) throws Exception {
```

Comment:
I'd make all of the protected methods `final` if we don't intend to override them. Just a little bit of extra signalling for the reader.